bin/render-activity-stream-html.js

```diff
@@ -0,0 +1,59 @@
+const fs = require("fs");
+const path = require("path");
+const prerender = require("./prerender");
+
+const BASE_FILE_PATH = path.resolve(__dirname, "../system-addon/data/content/");
+const PRERENDERED_HTML_FILE_PATH = path.join(BASE_FILE_PATH, "activity-stream-prerendered.html");
+const HTML_FILE_PATH = path.join(BASE_FILE_PATH, "activity-stream.html");
+const INITIAL_STATE_JS_FILE_PATH = path.join(BASE_FILE_PATH, "activity-stream-initial-state.js");
+
+const DEFAULT_OPTIONS = {
+  baseUrl: "resource://activity-stream/",
+  locale: "", // TODO: pass in options.locale. For now, we're just doing empty strings.
+  direction: "ltr",
+  title: "New Tab"
+};
+
+function templateHTML(options, html) {
+  const isPrerendered = !!html;
+  return `<!doctype html>
+<html lang="${options.locale}" dir="${options.direction}">
+  <head>
+    <meta charset="utf-8">
+    <meta http-equiv="Content-Security-Policy-Report-Only" content="script-src 'unsafe-inline'; img-src http: https: data: blob:; style-src 'unsafe-inline'; child-src 'none'; object-src 'none'; report-uri https://tiles.services.mozilla.com/v4/links/activity-stream/csp">
+    <title>${options.title}</title>
+    <link rel="icon" type="image/png" id="favicon" href="chrome://branding/content/icon32.png"/>
+    <link rel="stylesheet" href="chrome://browser/content/contentSearchUI.css" />
+    <link rel="stylesheet" href="${options.baseUrl}data/content/activity-stream.css" />
+    <link rel="icon" type="image/svg+xml" href="${options.baseUrl}img/newtab-icon.svg">
+  </head>
+  <body>
+    <div id="root">${isPrerendered ? html : ""}</div>${isPrerendered ? `
```

Comment:
Having so much of the semantics come from punctuation, and specifically nested template strings, makes this a bit hard to follow.  Pull out the <script> for stream-initial-state into a const higher up in the script?